src/gotcha_dl.c

```diff
@@ -0,0 +1,36 @@
+#include "gotcha_dl.h"
+#include "tool.h"
+#include "libc_wrappers.h"
+#include "elf_ops.h"
+
+void* dlopen_wrapper(const char* filename, int flags){
+  void* handle = orig_dlopen(filename,flags);
+  filterLibrariesByName("libdl");
+  struct binding_t* tool_iter = get_bindings();
+  for(;tool_iter!=NULL;tool_iter = tool_iter->next_binding){
+    /**
+     * If you remove this comparison, you will overwrite
+     * libdl's dlsym relocation with gotcha's, leading to
+     * infinite recursion. This is cool, but inefficient.
+     */
+    if(gotcha_strcmp(tool_iter->tool->tool_name, "gotcha")!=0){
+      gotcha_wrap(tool_iter->user_binding,tool_iter->user_binding_size,tool_iter->tool->tool_name);
```

Comment:
Wait a tic, I just learned to read again, there's something desperately wrong in this code. Momento, wow that's broken. Good catch